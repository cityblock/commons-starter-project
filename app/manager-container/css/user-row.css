@import '../../shared/css/sizes';

.container {
  background: white;
  padding: 12px 15px;
  display: flex;
  justify-content: space-between;
  flex-direction: row;
  align-items: center;
  border-radius: 5px;
  border: 1px solid #fff;
  color: var(--black-color);
  margin: 0 10px 10px 10px;
}

.title {
  font-weight: var(--roboto-medium-weight);
}

.meta {
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
  align-items: center;
}

.userDel {
  display: flex;
  justify-content: flex-end;
}

.dateSection {
  border-left: 1px solid var(--gray-background-color);
}

.dueAtSection {
  min-width: 97px;
  text-align: right;
}

.dateSection:last-child {
  padding-right: 0;
}

.dateLabel {
  font-family: var(--roboto);
  font-size: var(--extra-small-font-size);
  line-height: var(--extra-small-font-line-height);
  color: var(--light-gray-font-color);
  margin-right: 10px;
}

.dateValue {
  font-family: var(--roboto);
  font-size: var(--extra-small-font-size);
  line-height: var(--extra-small-font-line-height);
  color: var(--black-color);
}

.button {
  @apply --button;

  display: block;
  width: 180px;
  padding: 7px 0 3px;
  margin-bottom: 10px;
}

.buttonInverted {
  @apply --inverted-button;

  display: block;
  width: 180px;
  padding: 7px 0 3px;
  margin-bottom: 10px;
}

.popupHeading {
  font-size: var(--large-font-size);
  line-height: var(--large-font-line-height);
  text-align: center;
  margin-bottom: 50px;
}

.buttons {
  display: flex;
  justify-content: space-between;
}

.select {
  min-width: 100px;
}
