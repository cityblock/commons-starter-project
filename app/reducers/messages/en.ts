import { Lang } from '../locale-reducer';

export const ENGLISH_TRANSLATION = {
  lang: 'en' as Lang,
  messages: {
    'lightbox.of': 'of',
    'lightbox.image': 'image',
    'lightbox.images': 'images',
    'login.logInGoogle': 'Sign in with Google',
    'patientPanel.header': 'Patient Panel',
    'patientPanel.addPatient': 'Add Patient',
    'patientPanel.firstName': 'First Name',
    'patientPanel.lastName': 'Last Name',
    'patientPanel.age': 'Age',
    'patientPanel.location': 'Location',
    'patientPanel.joinedAt': 'Patient Joined',
    'patientPanel.engagedAt': 'Last Update',
    'patient.dateOfBirth': 'Date of birth',
    'patient.language': 'Preferred language',
    'patient.location': 'Location',
    'patient.joinedAt': 'Patient since',
    'patient.tasks': 'Tasks',
    'patient.timeline': 'Timeline',
    'patient.threeSixty': '360° View',
    'patient.map': 'MAP',
    'patient.context': 'Context',
    'patient.activity': 'Activity',
    'patient.submitProgressNote': 'Sign and Submit',
    'patient.activeCarePlan': 'Active',
    'patient.inactiveCarePlan': 'Inactive',
    'patient.medicaid': 'Medicaid',
    'patient.carePlanSuggestions': 'Suggestions',
    'patient.cbhNumber': 'CBH MRN#',
    'patient.patientInfo': 'Patient Info',
    'patient.createdPopupHeading': 'successfully enrolled',
    'patient.createdPopupBody':
      'Continue to their patient profile or go back to your patient roster.',
    'patient.createdPopupRoster': 'Patient Roster',
    'patient.createdPopupProfile': 'Go to profile',
    'encounter.new': 'Record new encounter',
    'encounter.newReason': 'Encounter reason',
    'encounter.newLocation': 'Location of encounter',
    'encounter.newStartTime': 'Start time',
    'encounter.newSummary': 'Encounter summary',
    'encounter.newAttachment': 'Add attachment',
    'encounter.newCancel': 'Cancel',
    'encounter.newSubmit': 'Submit',
    'error.tableLoadingHeading': 'Unable to load your roster',
    'error.tableLoadingMessage':
      'Sorry, something went wrong. Please try reloading the page again.',
    'error.tableLoadingButton': 'Try Again',
    'error.tableLoading': 'Loading',
    'backLink.back': 'Back',
    'careTeam.slack': 'Message on Slack',
    'careTeam.email': 'Send an email',
    'computedField.flag': 'Flag for review',
    'computedField.flagDetail':
      'Provide the following information for this question to be reviewed',
    'computedField.reason': 'Reason for review:',
    'computedField.reasonDetail': 'Enter the reason for review...',
    'concernCreate.addConcern': 'Add concern',
    'concernCreate.cancel': 'Cancel',
    'concernCreate.detail': "Complete this form to add a concern to this member's MAP",
    'concernCreate.hideAll': 'Hide all concerns',
    'concernCreate.loading': 'Loading...',
    'concernCreate.noResults': 'No matching concerns found.',
    'concernCreate.placeholder': 'Search for a concern...',
    'concernCreate.selectLabel': 'Add a concern:',
    'concernCreate.showAll': 'See all possible concerns',
    'concernCreate.submit': 'Add concern',
    'concernCreate.title': 'Add a concern',
    'concernCreate.selectConcernType': 'Select a concern type',
    'concernCreate.selectConcernTypeLabel': 'Concern type',
    'concernDelete.description':
      'Deleting it will completely remove it from this patient’s record.',
    'concernDelete.menu': 'Delete Concern',
    'concernDelete.name': 'Concern to be deleted:',
    'concernDelete.title': 'Are you sure you want to delete this concern?',
    'concernStats.created': 'Created:',
    'concernStats.goals': 'Goals:',
    'concernStats.lastUpdated': 'Last update:',
    'concernStats.tasks': 'Tasks:',
    'dashboard.countLoading': 'Loading',
    'dashboard.lists': 'Member lists',
    'dashboard.notifications': 'Notifications:',
    'dashboard.noUrgentTasks': 'There are no patients with tasks due or notifications',
    'dashboard.suggestions': 'Pending MAP suggestions',
    'dashboard.tasks': 'Tasks due and notifications',
    'dashboard.tasksDue': 'Tasks due:',
    'dashboard.team': 'New to your care team',
    'forms.optional': 'optional',
    'forms.declined': 'Declined',
    'forms.yes': 'Yes',
    'forms.no': 'No',
    'forms.clickHere': 'click here',
    'dateInfo.created': 'Created:',
    'dateInfo.updated': 'Updated:',
    'demographicsForm.dateOfBirth': 'Date of birth',
    'demographicsForm.firstName': 'First name',
    'demographicsForm.middleName': 'Middle name',
    'demographicsForm.lastName': 'Last name',
    'demographicsForm.maritalStatus': 'Marital status',
    'demographicsForm.maritalStatusPlaceholder': 'Select status',
    'demographicsForm.language': 'Preferred language',
    'demographicsForm.languagePlaceholder': 'Select language',
    'demographicsForm.gender': 'Gender',
    'demographicsForm.genderPlaceholder': 'Select gender',
    'demographicsForm.race': 'Race',
    'demographicsForm.racePlaceholder': 'Select race',
    'demographicsForm.ssn': 'Social security number',
    'demographicsForm.zip': 'Zip code',
    'contactForm.email': 'Email address',
    'contactForm.homePhone': 'Home phone number',
    'contactForm.mobilePhone': 'Mobile phone number',
    'contactForm.consentToText': 'Does the patient consent to being contacted via text?',
    'contactForm.consentToPhone': 'Does the patient consent to being contacted via phone?',
    'contactForm.readConsentStatement': 'Read consent statement to patient:',
    'screeningTool.submit': 'Submit',
    'screeningTool.resultsTitle': 'New Care Plan Suggestions',
    'screeningTool.resultsBody':
      "Based on the results of this tool, the following have been recommended as additions to the patient's care plan.",
    'goalCreate.addGoal': 'Add a goal',
    'goalCreate.back': 'Back',
    'goalCreate.cancel': 'Cancel',
    'goalCreate.custom': 'Write a new goal',
    'goalCreate.detail': "Choose from a list of goals or write a custom goal for this member's MAP",
    'goalCreate.goalAdded': 'Goal added:',
    'goalCreate.hideAll': 'Hide list of goals',
    'goalCreate.loading': 'Loading...',
    'goalCreate.noResults': 'No matching goals found. Adding as a custom goal.',
    'goalCreate.search': 'Search for a goal or write a custom goal...',
    'goalCreate.selectGoal': 'Select goal',
    'goalCreate.selectLabel': 'Add a goal:',
    'goalCreate.showAll': 'See list of goals',
    'goalCreate.submit': 'Add goal',
    'goalCreate.submitWithTasks': 'Done',
    'goalCreate.suggestedTasks': 'Suggested tasks:',
    'goalCreate.suggestionsDetail':
      'Based on this goal addition, the following suggestions have been recommended to this Member Action Plan. You must accept or dismiss these suggestions before continuing.',
    'goalCreate.suggestionsTitle': 'New MAP suggestions',
    'goalCreate.templates': 'All other goals',
    'goalCreate.title': 'Enter goal title:',
    'goalCreate.titlePlaceholder': 'Enter goal title...',
    'goalDelete.description': 'Deleting it will completely remove it from this patient’s record.',
    'goalDelete.menu': 'Delete Goal',
    'goalDelete.name': 'Goal to be deleted:',
    'goalDelete.title': 'Are you sure you want to delete this goal?',
    'history360.administered': 'Administered by:',
    'history360.conducted': 'Conducted:',
    'history360.noRecord': 'No record',
    'history360.previous': 'Previous score:',
    'insuranceForm.insuranceType': 'Insurance type',
    'insuranceForm.insuranceTypePlaceholder': 'Select insurance type',
    'insuranceForm.policyHolderRelationship': 'Patient relationship to policyholder',
    'insuranceForm.policyHolderRelationshipPlaceholder': 'Select relationship',
    'insuranceForm.memberId': 'Member ID',
    'insuranceForm.policyGroupNumber': 'Policy group number',
    'insuranceForm.issueDate': 'Issue date',
    'insuranceForm.expirationDate': 'Expiration date',
    'modalButtons.cancel': 'Cancel',
    'modalButtons.delete': 'Yes, delete',
    'modalButtons.submit': 'Submit',
    'patientInfo.jumpTo': 'Jump to:',
    'patientInfo.saveChanges': 'Save changes',
    'patientInfo.demographicInfo': 'Demographic information',
    'patientInfo.contactInfo': 'Contact information',
    'patientInfo.insuranceInfo': 'Insurance information',
    'patientMap.addGoal': 'Add a new goal',
    'patientMap.addTask': 'Add a task',
    'patientMap.nextUp': 'Next Up',
    'patientMap.emptyNextUpHeader': 'There are no Next Up concerns for this patient',
    'patientMap.emptyNextUpDetail':
      'Add concerns here that the patient and the care team do not want to focus on right now but would like to keep track of',
    'patientSearch.address': 'Home address',
    'patientSearch.dateOfBirth': 'Date of birth',
    'patientSearch.memberId': 'Member ID',
    'patientSearch.name': 'Patient name',
    'patientSearch.noResults': 'No results for this search',
    'patientSearch.noResultsDetail':
      'No matching patients for this search. Please check your spelling or try a new search.',
    'patientSearch.of': 'of',
    'patientSearch.placeholder': 'Search for member...',
    'patientSearch.resultsDescription': 'members found',
    'patientSearch.resultsDescriptionSingle': 'member found',
    'patientSearch.resultsTitle': 'Search results:',
    'patientSearch.resultsPlaceholder': 'Search for patients on your care team',
    'patientSearch.search': 'Search',
    'patientSearch.searchDescription': 'Search by first name or last name',
    'patientSearch.searchTitle': 'Search for member',
    'patientSearch.status': 'Status',
    'header.settings': 'Settings',
    'header.search': 'Search',
    'header.patients': 'Members',
    'header.tasks': 'Tasks',
    'header.logOut': 'Logout',
    'header.builder': 'Builder',
    'header.manager': 'Manager',
    'riskArea.assessmentType': 'Select assessment type',
    'riskArea.automated': 'Automated Assessment',
    'riskArea.highRiskThreshold': 'Enter assessment high risk threshold',
    'riskArea.manual': 'Manual Assessment',
    'riskArea.mediumRiskThreshold': 'Enter assessment medium risk threshold',
    'riskArea.order': 'Enter assessment order',
    'riskArea.title': 'Enter assessment title',
    'riskAreaGroup.close': 'Close',
    'riskAreaGroup.create': 'Create Domain',
    'riskAreaGroup.delete': 'Delete Domain',
    'riskAreaGroup.deleteDetail': 'Domain to be deleted:',
    'riskAreaGroup.deleteWarning': 'Are you sure you want to delete this domain?',
    'riskAreaGroup.empty': 'No Domains',
    'riskAreaGroup.highRiskThreshold': 'Enter domain high risk threshold',
    'riskAreaGroup.mediumRiskThreshold': 'Enter domain medium risk threshold',
    'riskArea.riskAreaGroupId': 'Select Domain',
    'riskAreaGroup.order': 'Enter domain order',
    'riskAreaGroup.shortTitle': 'Enter short domain title (radar chart)',
    'riskAreaGroup.title': 'Enter domain title',
    'riskAreaAssessment.administer': 'Administer tools',
    'riskAreaAssessment.save': 'Done',
    'riskAreaAssessment.start': 'Edit answers',
    'riskAreaAssessment.resultsTitle': 'New Care Plan Suggestions',
    'riskAreaAssessment.resultsBody':
      "Based on the results of this assessment, the following have been recommended as additions to the patient's care plan.",
    'select.default': 'Select one',
    'settings.heading': 'Settings',
    'settings.save': 'Save',
    'tasks.assignedToPlaceholder': 'Assign to',
    'tasks.patientGoalPlaceholder': 'Patient goal',
    'tasks.listView': 'Tasks',
    'tasks.createTask': 'Create a new task',
    'tasks.notifications': 'Notifications',
    'tasks.noTasks': 'No Tasks',
    'taskCreate.addTask': 'Add a task',
    'taskCreate.assignee': 'Assign task to:',
    'taskCreate.cancel': 'Cancel',
    'taskCreate.concern': 'Concern:',
    'taskCreate.description': 'Enter task description:',
    'taskCreate.descriptionPlaceholder': 'Enter task description...',
    'taskCreate.detail': 'Complete this form to add a task to this goal',
    'taskCreate.dueAt': 'Due date:',
    'taskCreate.goal': 'Goal:',
    'taskCreate.priority': 'Task priority:',
    'taskCreate.submit': 'Add task',
    'taskCreate.title': 'Enter task title:',
    'taskCreate.titlePlaceholder': 'Enter task title...',
    'taskComment.activity': 'Activity and comments',
    'taskComment.error': 'Error adding comment',
    'taskDelete.cancel': 'Cancel',
    'taskDelete.confirm': 'Yes, delete',
    'taskDelete.title': 'Are you sure you want to delete this task?',
    'taskDelete.titleError': 'Error deleting task.',
    'taskDelete.body': "Deleting this task will completely remove it from this patient's record.",
    'taskDelete.bodyError': 'Please try again.',
    'taskDescription.empty': 'Click to add a task description...',
    'task.assign': 'Assign Task:',
    'task.due': 'Due',
    'task.followers': 'Followers of this task:',
    'task.opened': 'Opened',
    'task.noDueDate': 'No due date',
    'task.priority': 'Task priority:',
    'taskPriority.low': 'Low priority',
    'taskPriority.medium': 'Medium priority',
    'taskPriority.high': 'High priority',
    'taskPriority.select': 'Select priority',
    'textarea.default': 'Enter response...',
    'threeSixty.automated': 'Automated Assessments:',
    'threeSixty.automatedDetail': 'Automated Assessments',
    'threeSixty.automatedEmpty': 'There are no automated assessments on record',
    'threeSixty.back': 'Back to 360',
    'threeSixty.history': 'History',
    'threeSixty.historyEmpty': 'No screening tools have been administered to this patient',
    'threeSixty.initialAssessment': 'Initial Assessment:',
    'threeSixty.manual': 'Manual Assessments:',
    'threeSixty.manualDetail': 'Manual Assessments',
    'threeSixty.manualEmpty': 'There are no manual assessments on record',
    'threeSixty.never': 'Not submitted',
    'threeSixty.noAssessments': 'No assessments on record',
    'threeSixty.notCompleted': 'Member has not completed this assessment.',
    'threeSixty.notCompletedShort': 'Not completed',
    'threeSixty.summary': 'Patient Summary',
    'notifications.leftPane': 'Left Pane',
    'notifications.noNotifications': 'No Notifications',
    'notification.time': 'Time',
    'user.createdAt': 'Created At',
    'user.editedAt': 'Last edited',
    'user.delete': 'Delete User',
    'progressNote.new': 'New progress note',
    'progressNote.progressNotes': 'Progress Note',
    'progressNote.selectType': 'Select progress note type:',
    'progressNote.selectTime': 'Select progress note time:',
    'progressNote.selectLocation': 'Select progress note location:',
    'progressNote.contextAndPlan': 'Context and plan',
    'progressNote.memberConcernAndObservation': 'Member concern and observation',
    'progressNote.numberMapUpdates': 'MAP updates',
    'progressNote.goToActivity': 'go to activity',
    'progressNote.updateMap': 'Update MAP',
    'progressNote.update360': 'Update 360',
    'progressNote.administerTool': 'Administer tool',
    'progressNote.doesRequireCosignature': 'Does this progress note require supervisor review?',
    'progressNote.selectSupervisor': 'Select supervisor:',
    'progressNote.supervisorReview': 'Supervisor Note',
    'progressNote.pendingReview': 'Pending review',
    'progressNote.reviewed': 'Reviewed',
    'quickCallNote.new': '+ Quick call note',
    'quickCallForm.title': 'Quick call note',
    'quickCallForm.submit': 'Save call',
    'quickCallForm.inboundCall': 'Inbound call',
    'quickCallForm.outboundCall': 'Outbound call',
    'quickCallForm.startTime': 'What time was the call?',
    'quickCallForm.callRecipient': 'Who was the call with?',
    'quickCallForm.wasSuccessful': 'Were you able to speak with them?',
    'quickCallForm.reason': 'What was the reason for the call?',
    'quickCallForm.summary': 'Please provide additional notes about the call here …',
    'computedField.create': 'Create Computed Field',
    'computedFieldCreate.createLabelPlaceholder': 'Enter label',
    'computedFieldCreate.createDataTypeLabel': 'Select a data type',
    'computedFieldCreate.cancel': 'Cancel',
    'computedField.confirmDelete': 'Yes, delete',
    'question.selectComputedField': 'Set computed field',
    'question.notComputedField': 'Not a computed field',
    'question.selectApplicable': 'Select an applicable if type (required!)',
    'question.applicableOneTrue': 'One true',
    'question.applicableAllTrue': 'All true',
    'question.selectAnswerType': 'Select an answer type (required!)',
    'question.answerTypeDropdown': 'Dropdown',
    'question.answerTypeRadio': 'Radio',
    'question.answerTypeFreeText': 'Free Text',
    'question.answerTypeMultiselect': 'Multiselect',
    'question.selectHasOtherTextAnswer': 'Should this question have an "other" answer?',
    'question.hasOtherTextAnswerTrue': 'Yes',
    'question.hasOtherTextAnswerFalse': 'No',
    'questionConditionCreate.selectAnswer': 'Select Answer',
    'carePlanSuggestions.seeSuggestions': 'See Suggestions',
    'riskAdjustmentType.inactive': 'inactive',
    'riskAdjustmentType.increment': 'increment',
    'riskAdjustmentType.forceHighRisk': 'force high risk',
    'riskAdjustmentType.selectRiskAdjustmentType': 'Select risk adjustment type',
    'builder.createConcern': 'Create Concern',
    'builder.createAssessment': 'Create Assessment',
    'builder.createQuestion': 'Create Question',
    'builder.createGoal': 'Create Goal',
    'builder.createProgressNoteTemplate': 'Create Progress Note Template',
    'builder.domains': 'Domains',
    'builder.assessments': 'Assessments',
    'builder.questions': 'Questions',
    'builder.concerns': 'Concerns',
    'builder.goals': 'Goals',
    'builder.tools': 'Tools',
    'builder.progressNoteTemplates': 'Progress Note Templates',
    'builder.computedFields': 'Computed Fields',
    'builder.createScreeningTool': 'Create Screening Tool',
    'manager.inviteUser': 'Invite User',
    'manager.users': 'Users',
    'manager.invites': 'Invites',
  },
};
