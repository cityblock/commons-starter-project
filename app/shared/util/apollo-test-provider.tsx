import { InMemoryCache } from 'apollo-cache-inmemory';
import { ApolloClient } from 'apollo-client';
import { SchemaLink } from 'apollo-link-schema';
import { print } from 'graphql';
import { addMockFunctionsToSchema, makeExecutableSchema } from 'graphql-tools';
import createHistory from 'history/createBrowserHistory';
import * as React from 'react';
import { ApolloProvider } from 'react-apollo';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import * as schema from '../../../server/graphql/schema.graphql';
import ReduxConnectedIntlProvider from '../../redux-connected-intl-provider';
import createStore from '../../store';

const history = createHistory();
const store = createStore(history);

interface IProps {
  graphqlMocks: any;
}

// Helper in our codebase to create the apollo client
// from http://blog.dideric.is/2018/03/18/Testing-apollo-containers/
export default class ApolloTestProvider extends React.Component<IProps> {
  static createSchema = () => makeExecutableSchema({ typeDefs: print(schema) });

  apolloClient: any;
  schema: any;

  constructor(props: IProps) {
    super(props);
    // Every instance should have it's own schema instance so tests
    // don't bleed into one another
    this.schema = ApolloTestProvider.createSchema();
    this.apolloClient = new ApolloClient<any>({
      link: new SchemaLink({ schema: this.schema }),
      cache: new InMemoryCache().restore({}) as any,
    });
    this.addDefaultMocks();
  }

  componentWillMount() {
    const { graphqlMocks } = this.props;
    this.mockGraphql(graphqlMocks);
  }

  componentWillReceiveProps({ graphqlMocks }: any) {
    this.mockGraphql(graphqlMocks);
  }

  addDefaultMocks() {
    addMockFunctionsToSchema({
      schema: this.schema,
    });
  }

  mockGraphql(mocks = {}) {
    addMockFunctionsToSchema({
      schema: this.schema,
      mocks,
    });
  }

  render() {
    const { children } = this.props;
    return (
      <Provider store={store}>
        <ApolloProvider client={this.apolloClient}>
          <ReduxConnectedIntlProvider defaultLocale="en">
            <MemoryRouter>{children}</MemoryRouter>
          </ReduxConnectedIntlProvider>
        </ApolloProvider>
      </Provider>
    );
  }
}
