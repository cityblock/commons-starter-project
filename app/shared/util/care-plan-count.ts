import { FullCarePlanSuggestionFragment } from '../../graphql/types';

export function getConcernCount(carePlanSuggestions: FullCarePlanSuggestionFragment[]) {
  if (!carePlanSuggestions.length) {
    return 0;
  }

  const concernSuggestions = carePlanSuggestions.filter(
    suggestion => suggestion.suggestionType === 'concern',
  );

  return concernSuggestions.length;
}

export function getGoalSuggestions(carePlanSuggestions: FullCarePlanSuggestionFragment[]) {
  if (!carePlanSuggestions.length) {
    return [];
  }

  return carePlanSuggestions.filter(suggestion => suggestion.suggestionType === 'goal');
}

export function getGoalCount(carePlanSuggestions: FullCarePlanSuggestionFragment[]) {
  return getGoalSuggestions(carePlanSuggestions).length;
}

export function getTaskCount(carePlanSuggestions: FullCarePlanSuggestionFragment[]) {
  if (getGoalCount(carePlanSuggestions) === 0) {
    return 0;
  }

  const goalSuggestions = getGoalSuggestions(carePlanSuggestions);

  const taskSuggestions = goalSuggestions
    .map(goalSuggestion => goalSuggestion.goalSuggestionTemplate!.taskTemplates)
    .reduce((taskSuggestions1, taskSuggestions2) => taskSuggestions1.concat(taskSuggestions2));

  return (taskSuggestions || []).length;
}
