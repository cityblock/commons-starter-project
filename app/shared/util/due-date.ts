export const checkIfDueSoon = (dateString: string | null) => {
  let dueSoon = false;

  if (dateString) {
    const taskDueDate = new Date(dateString);
    dueSoon = taskDueDate.valueOf() < Date.now() + 60 * 60 * 24 * 1000;
  }

  return dueSoon;
};
