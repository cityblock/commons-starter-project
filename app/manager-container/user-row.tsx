import * as React from 'react';
import { FormattedDate, FormattedMessage, FormattedRelative } from 'react-intl';
import { FullUserFragment, Permissions } from '../graphql/types';
import Button from '../shared/library/button/button';
import Option from '../shared/library/option/option';
import Select from '../shared/library/select/select';
import { Popup } from '../shared/popup/popup';
import withCurrentUser, { IInjectedProps } from '../shared/with-current-user/with-current-user';
import * as styles from './css/user-row.css';

interface IProps extends IInjectedProps {
  user: FullUserFragment;
  deleteUser: (userEmail: string) => void;
  editUserPermissions: (permissions: Permissions, userEmail: string) => void;
}

interface IState {
  popupVisible: boolean;
}

export class UserRow extends React.Component<IProps, IState> {
  constructor(props: IProps) {
    super(props);

    this.onCloseClick = this.onCloseClick.bind(this);
    this.onOpenClick = this.onOpenClick.bind(this);
    this.onConfirmClick = this.onConfirmClick.bind(this);

    this.state = {
      popupVisible: false,
    };
  }

  formatUser() {
    const { user } = this.props;
    return `${user.firstName || ''} ${user.lastName || ''} (${user.email})`;
  }

  onCloseClick() {
    this.setState({ popupVisible: false });
  }

  onOpenClick() {
    this.setState({ popupVisible: true });
  }

  onConfirmClick() {
    const { user } = this.props;
    if (user.email) {
      this.props.deleteUser(user.email);
      this.setState({ popupVisible: true });
    }
  }

  renderPopup() {
    const { popupVisible } = this.state;
    return (
      <Popup visible={popupVisible}>
        <div className={styles.popupHeading}>
          Are you sure you want to delete {this.formatUser()}?
        </div>
        <div className={styles.buttons}>
          <div className={styles.buttonInverted} onClick={this.onCloseClick}>
            Close
          </div>
          <div className={styles.button} onClick={this.onConfirmClick}>
            Confirm Delete
          </div>
        </div>
      </Popup>
    );
  }

  render() {
    const { user, editUserPermissions, featureFlags } = this.props;
    const popup = this.renderPopup();
    const onChangeUserPermissions = (event: React.ChangeEvent<HTMLSelectElement>) => {
      editUserPermissions(event.target.value as Permissions, user.email!);
    };
    const formattedCreatedAt = (
      <FormattedRelative value={user.createdAt}>
        {(date: string) => <span className={styles.dateValue}>{date}</span>}
      </FormattedRelative>
    );
    const formattedUpdatedAt = (
      <FormattedDate value={user.updatedAt} year="numeric" month="short" day="numeric">
        {(date: string) => <span className={styles.dateValue}>{date}</span>}
      </FormattedDate>
    );
    return (
      <div className={styles.container}>
        <div className={styles.title}>{this.formatUser()}</div>
        <div className={styles.meta}>
          <div className={styles.dateSection}>
            {featureFlags.canChangeUserPermissions && (
              <Select
                value={user.permissions}
                onChange={onChangeUserPermissions}
                className={styles.select}
              >
                <Option value="green">Green</Option>
                <Option value="pink">pink</Option>
                <Option value="orange">Orange</Option>
                <Option value="blue">Blue</Option>
                <Option value="yellow">Yellow</Option>
                <Option value="red">Red</Option>
                <Option value="black">Black</Option>
              </Select>
            )}
          </div>
          <div className={styles.dateSection}>
            <FormattedMessage id="user.createdAt">
              {(message: string) => <span className={styles.dateLabel}>{message}:</span>}
            </FormattedMessage>
            {formattedCreatedAt}
          </div>
          <div className={styles.dateSection}>
            <FormattedMessage id="user.editedAt">
              {(message: string) => <span className={styles.dateLabel}>{message}:</span>}
            </FormattedMessage>
            {formattedUpdatedAt}
          </div>
          {featureFlags.canDeleteUsers && (
            <div className={styles.userDel}>
              <Button messageId="user.delete" onClick={this.onOpenClick} />
            </div>
          )}
          {popup}
        </div>
      </div>
    );
  }
}

export default withCurrentUser()(UserRow);
