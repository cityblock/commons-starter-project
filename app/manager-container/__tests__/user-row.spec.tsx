import { createMemoryHistory } from 'history';
import * as React from 'react';
import { MockedProvider } from 'react-apollo/test-utils';
import { ConnectedRouter } from 'react-router-redux';
import { create } from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import { ENGLISH_TRANSLATION } from '../../reducers/messages/en';
import ReduxConnectedIntlProvider from '../../redux-connected-intl-provider';
import { user } from '../../shared/util/test-data';
import { UserRow } from '../user-row';

const oldDate = Date.now;

describe('user row', () => {
  beforeAll(() => {
    Date.now = jest.fn(() => 1500494779252);
  });
  afterAll(() => {
    Date.now = oldDate;
  });

  it('renders manager users with users', () => {
    const mockStore = configureMockStore([]);
    const history = createMemoryHistory();
    const locale = { messages: ENGLISH_TRANSLATION.messages };
    const deleteUser = () => false;
    const tree = create(
      <MockedProvider mocks={[]} store={mockStore({ locale })}>
        <ReduxConnectedIntlProvider>
          <ConnectedRouter history={history}>
            <UserRow user={user} deleteUser={deleteUser} />
          </ConnectedRouter>
        </ReduxConnectedIntlProvider>
      </MockedProvider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
