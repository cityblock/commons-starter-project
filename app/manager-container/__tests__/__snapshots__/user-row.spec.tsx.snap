// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user row renders manager users with users 1`] = `
<div
  className="container"
>
  <div
    className="title"
  >
    first last (b@c.com)
  </div>
  <div
    className="meta"
  >
    <div
      className="dateSection"
    />
    <div
      className="dateSection"
    >
      <span
        className="dateLabel"
      >
        Created At
        :
      </span>
      <span
        className="dateValue"
      >
        in 2 months
      </span>
    </div>
    <div
      className="dateSection"
    >
      <span
        className="dateLabel"
      >
        Last edited
        :
      </span>
      <span
        className="dateValue"
      >
        Sep 7, 2017
      </span>
    </div>
  </div>
</div>
`;
